app/assets/purs/src/EntryPoints/Comments.purs

```diff
@@ -99,3 +101,7 @@ def props =
             { onClick: dispatch Create } 
             "Create"
         ]   
+
+createComment :: Int -> String -> String -> Aff Unit
+createComment = API.post "create_comment_path" \call postId commenter comment -> 
+  call { postId: postId, commenter, comment } >>= API.ignoreResponse
```

Comment:
It seems like most APIs I found in the main repo did .ignoreResponse - is it safe to assume that if I had some server-side validations (like comment body must be > 20 characters or user can only make 1 comment on a post), I'd probably have to worry about the API response? Or do we generally only care about client-side validations?

Reply:
Another flavor of the same question: in this code, I make the API call and update the component's state. Is there ever a need to sync up the state with the backend? For example, instead snoc-ing to the list of comments of my state, would I ever want to do something like `state.comments = API.getComments`?

Reply:
> It seems like most APIs I found in the main repo did .ignoreResponse - is it safe to assume that if I had some server-side validations (like comment body must be > 20 characters or user can only make 1 comment on a post), I'd probably have to worry about the API response? Or do we generally only care about client-side validations?

Yeah, exactly. The `ignoreResponse` pattern isn't ideal, as in a lot of cases we don't even validate request success before updating the `state` accordingly, but in theory that is certainly a situation in which you'd need to worry about the response. 

Another case where we parse the request responses are when we lazy load data (like into Hub) to improve perceived load times.

Reply:
> Another flavor of the same question: in this code, I make the API call and update the component's state. Is there ever a need to sync up the state with the backend? For example, instead snoc-ing to the list of comments of my state, would I ever want to do something like `state.comments = API.getComments`?

In general, we would do something like what you did.  If you did need to sync them up (say when you write a record to the database and need to get it's ID back), you would parse the API response and then update your `state`

Reply:
Makes sense, thanks for answering!